### DISCUSSION TITLE
Logging/logger is printing to Jupyter and Spyder/IPython Console during lasio.read()

### DISCUSSION THREAD

**Discussion by bzlmnop** [kinverarity1/lasio#497] [Q&A] [answered]
I have been using lasio for a few months now. 
I have one jupyter notebook that I use it in to read a bunch of .las files. 
I just started a new notebook doing a similar operation, and now every time lasio.read() is called, it prints "Opening {my las file} as ascii and treating errors with "replace"" where {my las file} is the file path string I passed to the function. 
I don't recall this happening in my previous notebook and cannot figure out how to prevent it from happening. 
It is seriously slowing down my code. 
Any ideas/help would be greatly appreciated.

**Comment by ThomasMGeo** [comment]
What version of lasio?

On Wed, Nov 3, 2021 at 2:04 PM bzlmnop ***@***.***> wrote:

> I have been using lasio for a few months now.
> I have one jupyter notebook that I use it in to read a bunch of .las files.
> I just started a new notebook doing a similar operation, and now every
> time lasio.read() is called, it prints "Opening {my las file} as ascii and
> treating errors with "replace"" where {my las file} is the file path string
> I passed to the function.
> I don't recall this happening in my previous notebook and cannot figure
> out how to prevent it from happening.
> It is seriously slowing down my code.
> Any ideas/help would be greatly appreciated.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/kinverarity1/lasio/discussions/497>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJDVBLCOROCM5O3ZTMU2H63UKGPXRANCNFSM5HJ2ISGQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
Thomas Martin
(805) 904-9991

**Reply by bzlmnop** [reply]
Current version 0.29. 
I think I solved it by adding:
`import logging`
`logging.basicConfig()`
`logging.getLogger().setLevel(logging.ERROR)`
For Jupyter, this code has to be in the same cell as where you are calling lasio.read().
I tried using ipython magic commands such as: 
`%config Application.log_level='ERROR'` 
Which, did not work in either Jupyter or Spyder.

**Comment by bzlmnop** [comment, marked as answer]
For me the fix was to add:
`import logging`
`logging.basicConfig()`
`logging.getLogger().setLevel(logging.ERROR)`
To the cell in Jupyter or the script in Spyder that is calling lasio.read().

**Comment by dcslagel** [comment]
Looks like you found the right solution.  

For reference, there is a discussion and examples of setting logging in the "set logging level for lasio" example notebook at:
https://github.com/kinverarity1/lasio/blob/master/notebooks/set%20logging%20level%20for%20lasio.ipynb

It covers configuring logging to add more logging output and, toward the end, how to reduce logging output.
